scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,331 @@
+            bugzilla_password = getpass(prompt="Password: ")
+            authentication = {"username": bugzilla_username, "password:": bugzilla_password}
+            resp = send_data(bug_id, archive_name, file_data, comment, authentication)
+            resp_json = resp.json()
+        # 101: Invalid bug id
+        elif resp_json["code"] == 101:
+            print("Invalid bug id")
+            new_bug_id = input("Enter a new bug id (leave blank to exit): ")
+            if new_bug_id == "":
+                print("ID left blank, exiting")
+                exit(0)
+            bug_id, valid = try_parse_int(new_bug_id)
+            # Try and see if the new supplied ID is a positive integer
+            while not valid or bug_id <= 0:
+                print("Could not parse bug id as valid, try again")
+                new_bug_id = input("Enter a new bug id (leave blank to exit): ")
+                if new_bug_id == "":
+                    print("ID left blank, exiting")
+                    exit(0)
+                bug_id, valid = try_parse_int(new_bug_id)
+            resp = send_data(bug_id, archive_name, file_data, comment, authentication)
+            resp_json = resp.json()
+        else:
+            print("Error: " + resp_json["message"])
+            exit(1)
+    print("File successfully uploaded to Bugzilla")
+
+def run_must_gather(image, logfolder):
+    # If the log folder already exists, delete it
+    if os.path.isdir(logfolder):
+        shutil.rmtree(logfolder)
+
+    # Make a new log folder
+    os.mkdir(logfolder)
+
+    # Open the output file
+    with open(logfolder + OUTPUT_FILE, "w+") as out_file:
+        # Run oc adm must-gather with the appropriate image and dest-dir
+        print("Running must-gather")
+        try:
+            subprocess.run(
+                ["oc", "adm", "must-gather",
+                "--image=" + image, "--dest-dir=" + logfolder],
+                stdout=out_file, stderr=subprocess.PIPE, check=True)
+        except subprocess.CalledProcessError as e:
+            print("Error in the execution of must-gather: " + e.stderr)
+            exit(1)
+
+
+def trim_logs(logfolder, num_seconds):
+    for subdir, _, files in os.walk(logfolder):
+        for file in files:
+            if file == "must-gather.log": #Ignore the log made by capturing the output of must-gather
+                continue
+            if ".log" in file:
+                trim_file_by_time(os.path.join(subdir, file), num_seconds, PODLOG_TIMESTAMP_REGEX, PODLOG_TIMESTAMP_FORMAT)
+            if "kubelet" in file or "NetworkManager" in file:
+                trim_file_by_time(os.path.join(subdir, file), num_seconds, NODELOG_TIMESTAMP_REGEX, NODELOG_TIMESTAMP_FORMAT)
+
+def try_parse_int(value):
+    """Tries to parse the value as an int"""
+    try:
+        return int(value), True
+    except ValueError:
+        return value, False
+
+
+
+def send_data(bug_id, file_name, file_data, comment, authentication):
```

Comment:
perhaps we should check the file size before attaching it to make sure it is under 19.5MB ?

Reply:
Done in recent commit